airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -947,9 +947,7 @@ def execute(self, context) -> None:
                 delegate_to=self.delegate_to,
                 impersonation_chain=self.impersonation_chain,
             )
-            schema_fields = json.loads(
-                gcs_hook.download(gcs_bucket, gcs_object).decode("utf-8")  # type: ignore[attr-defined]
-            )  # type: ignore[attr-defined]
+            schema_fields = json.loads(gcs_hook.download(gcs_bucket, gcs_object))
```

Comment:
I got an error in CI. To fix it this modification is necessary. In this case GCSHook.download() returns always `bytes` and json.loads() calls decode internally.

If we preserve the call to `decode`, we need to cast the returned value from `download` to bytes. @turbaszek could you advise me about this point?

Reply:
Looks good to me. I'm surprised we didn't catch it earlier

Reply:
Thank you for your review. Static type checking is really cool because we can find like above the issues.

Reply:
Well, this never worked as per #12439 👀